tests/test_api.py

```diff
@@ -981,14 +985,140 @@ def test_empty_to_uuid(self):
+    """Unit tests for add_organization"""
+
+    def test_add_new_organization(self):
+        """Check if everything goes OK when adding a new organization"""
+
+        organization = api.add_organization(name='Example')
+
+        # Tests
+
+        self.assertIsInstance(organization, Organization)
+
+        self.assertEqual(organization.name, 'Example')
+
+        organizations_db = Organization.objects.filter(name='Example')
+        self.assertEqual(len(organizations_db), 1)
+
+        org1 = organizations_db[0]
+
+        self.assertEqual(organization, org1)
+
+    def test_add_duplicate_organization(self):
+        """Check if it fails when adding a duplicate organization"""
+
+        org = api.add_organization(name='Example')
+
+        with self.assertRaisesRegex(AlreadyExistsError, ORGANIZATION_ALREADY_EXISTS_ERROR.format(name=org.name)):
+            org = api.add_organization(name=org.name)
+
+        organizations = Organization.objects.filter(name='Example')
+        self.assertEqual(len(organizations), 1)
+
+    def test_organization_name_none(self):
+        """Check if it fails when organization name is `None`"""
+
+        with self.assertRaisesRegex(InvalidValueError, ORGANIZATION_NAME_NONE_OR_EMPTY_ERROR):
+            api.add_organization(name=None)
+
+    def test_organization_name_empty(self):
+        """Check if it fails when organization name is empty"""
+
+        with self.assertRaisesRegex(InvalidValueError, ORGANIZATION_NAME_NONE_OR_EMPTY_ERROR):
+            api.add_organization(name='')
+
+    def test_organization_name_whitespaces(self):
+        """Check if it fails when organization name is composed by whitespaces only"""
+
+        with self.assertRaisesRegex(InvalidValueError, ORGANIZATION_NAME_NONE_OR_EMPTY_ERROR):
+            api.add_organization(name='   ')
+
+        with self.assertRaisesRegex(InvalidValueError, ORGANIZATION_NAME_NONE_OR_EMPTY_ERROR):
+            api.add_organization(name='\t')
+
+        with self.assertRaisesRegex(InvalidValueError, ORGANIZATION_NAME_NONE_OR_EMPTY_ERROR):
+            api.add_organization(name=' \t  ')
+
+    def test_organization_name_int(self):
+        """Check if it fails when organization name is an integer"""
+
+        with self.assertRaisesRegex(TypeError, ORGANIZATION_VALUE_ERROR):
+            api.add_organization(name=12345)
+
+
+class TestDeleteOrganization(TestCase):
+    """Unit tests for delete_organization"""
+
+    def setUp(self):
+        """Load initial dataset"""
+        api.add_organization(name='Example')
+        api.add_organization(name='Bitergia')
+        api.add_organization(name='Libresoft')
+
+        jsmith = api.add_identity('scm', email='jsmith@example.com')
+        uidentity = api.enroll(jsmith.id, 'Example',
+                               from_date=datetime.datetime(1999, 1, 1),
+                               to_date=datetime.datetime(2000, 1, 1))
+
+    def test_delete_organization(self):
```

Comment:
This test should also check whether the other organizations weren't removed. The rationale is that the code might be wrong and removes random organizations or all the organizations.

Reply:
It makes total sense. I have added checks for this:

* Check number of organizations (2 orgs, after removing 1 out of 3).
* Check their names.